### PR TITLE
Facepile component, people picker updates, circle persona with initials

### PR DESCRIPTION
**Pull request by philworthington**
- New Facepile component
- New people picker functionality for adding people to the facepile
- New persona initials and sizes (doughboy placeholder removed)

### CODE REVIEW

File: src/components/Facepile/Facepile.html [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+<div class="ms-FacePile">
+  <button class="ms-FacePile-itemBtn ms-FacePile-itemBtn--addPerson js-addPerson">
+    <i class="ms-FacePile-addPersonIcon ms-Icon ms-Icon--personAdd"></i>
+  </button>
+  <div class="ms-FacePile-members">
+    <button class="ms-FacePile-itemBtn ms-FacePile-itemBtn--member" title="Alton Lafferty">
+      <div class="ms-Persona ms-Persona--xs">
+        <div class="ms-Persona-imageArea">
+          <div class="ms-Persona-initials ms-Persona-initials--blue">AL</div>
+          <img class="ms-Persona-image" src="../persona/Persona.Person2.png">
```
Comment (notandrew) [inline src/components/Facepile/Facepile.html]: We should have alt text here.

File: src/components/Facepile/Jquery.Facepile.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      var $pickerSearchField = $picker.find(".ms-PeoplePicker-searchField");
+      var $personaCard = $('.ms-FacePile').find('.ms-PersonaCard');
+
+
+      /** Increment member count and show/hide overflow text */
+      var incrementMembers = function() {
+        /** Increment person count by one */
+        $membersCount += 1;
+
+        /** Display a maxiumum of 5 people */
```
Comment (notandrew) [inline src/components/Facepile/Jquery.Facepile.js]: sp: maximum

File: src/components/Facepile/Jquery.Facepile.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+        var $facePileMember = $membersList.find(".ms-Persona-primaryText:contains(" + memberText + ")").first();
+
+        if ($facePileMember) {
+          $facePileMember.parent().closest('.ms-FacePile-itemBtn').remove();
+
+          $membersCount -= 1;
+
+           /** Display a maxiumum of 5 people */
+           $(".ms-FacePile-members").children(":lt(5)").show();
```
Comment (notandrew) [inline src/components/Facepile/Jquery.Facepile.js]: We're using a lot of hard-coded 5s.  Would it make sense to put this into a variable?

File: src/components/PeoplePicker/PeoplePicker.less [hunk 4]
```diff
@@ -176,33 +180,51 @@
   margin: 0 0 10px 0;
```
Comment (notandrew) [inline src/components/PeoplePicker/PeoplePicker.less]: can this be changed to margin-bottom or is there a reason to keep the other coordinates? (same with the padding)

**Comment by notandrew** [comment]
looks good

**Comment by mikewheaton** [comment]
Any chance of breaking this up into multiple pull requests? It's a lot to review.

File: gulp/fabric.js [hunk 5]
```diff
@@ -0,0 +1,842 @@
+var gulp = require('gulp');
```
Comment (battletoilet) [inline gulp/fabric.js]: Make sure we remove this and our conflict resolution changes are in.

File: src/components/Facepile/Facepile.html [hunk 6]
```diff
@@ -0,0 +1,252 @@
+<!-- Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE in the project root for license information. -->
+
+<div class="ms-Facepile">
+  <button class="ms-Facepile-itemBtn ms-Facepile-itemBtn--addPerson js-addPerson">
+    <i class="ms-Facepile-addPersonIcon ms-Icon ms-Icon--personAdd"></i>
+  </button>
+  <div class="ms-Facepile-members">
+    <button class="ms-Facepile-itemBtn ms-Facepile-itemBtn--member" title="Alton Lafferty">
+      <div class="ms-Persona ms-Persona--xs">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: <div> Cannot be used inside of <button>

File: src/components/Facepile/Facepile.html [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        </div>
+        <div class="ms-Persona-presence"></div>
+        <div class="ms-Persona-details">
+          <div class="ms-Persona-primaryText">Alton Lafferty</div>
+          <div class="ms-Persona-secondaryText">Accountant</div>
+        </div>
+      </div>
+    </button>
+    <button class="ms-Facepile-itemBtn ms-Facepile-itemBtn--member" title="Douglas Fielder">
+      <div class="ms-Persona ms-Persona--xs">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same comment as Above, `<div>` cannot be used inside of a `<button>`

File: src/components/Facepile/Facepile.html [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+          <img class="ms-Persona-image" src="">
+        </div>
+        <div class="ms-Persona-presence"></div>
+        <div class="ms-Persona-details">
+          <div class="ms-Persona-primaryText">Douglas Fielder</div>
+          <div class="ms-Persona-secondaryText">Public Relations</div>
+        </div>
+      </div>
+    </button>
+    <button class="ms-Facepile-itemBtn ms-Facepile-itemBtn--member" title="Marcus Lauer">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same comment as Above, `<div>` cannot be used inside of a `<button>`

File: src/components/Facepile/Facepile.html [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+          <input class="ms-PeoplePicker-searchField" type="text" placeholder="Add members">
+        </div>
+        <!-- Suggested list -->
+        <div class="ms-PeoplePicker-results">
+          <div class="ms-PeoplePicker-peopleListHeader">
+            <span>Suggested contacts</span>
+          </div>
+          <ul class="ms-PeoplePicker-peopleList">
+            <li class="ms-PeoplePicker-peopleListItem ms-PeoplePicker-result">
+              <button class="ms-PeoplePicker-peopleListBtn">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same comment as Above, `<div>` cannot be used inside of a `<button>`

File: src/components/Facepile/Facepile.html [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+                  <div class="ms-Persona-details">
+                    <div class="ms-Persona-primaryText">Russel Miller</div>
+                    <div class="ms-Persona-secondaryText">Sales</div>
+                  </div>
+                </div>
+              </button>
+            </li>
+            <li class="ms-PeoplePicker-peopleListItem ms-PeoplePicker-result">
+              <button class="ms-PeoplePicker-peopleListBtn">
+                <div class="ms-Persona ms-Persona--selectable ms-Persona--sm">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same comment as Above, `<div>` cannot be used inside of a `<button>`

File: src/components/Facepile/Facepile.html [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+                  <div class="ms-Persona-details">
+                    <div class="ms-Persona-primaryText">Douglas Fielder</div>
+                    <div class="ms-Persona-secondaryText">Public Relations</div>
+                  </div>
+                </div>
+              </button>
+            </li>
+            <li class="ms-PeoplePicker-peopleListItem ms-PeoplePicker-result">
+              <button class="ms-PeoplePicker-peopleListBtn">
+                <div class="ms-Persona ms-Persona--selectable ms-Persona--sm">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same comment as Above, `<div>` cannot be used inside of a `<button>`

File: src/components/Facepile/Facepile.html [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+                  <div class="ms-Persona-presence"></div>
+                  <div class="ms-Persona-details">
+                    <div class="ms-Persona-primaryText">Jessica Fischer</div>
+                    <div class="ms-Persona-secondaryText">Public Relations</div>
+                  </div>
+                </div>
+              </button>
+            </li>
+            <li class="ms-PeoplePicker-peopleListItem ms-PeoplePicker-result">
+              <button class="ms-PeoplePicker-peopleListBtn">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: Same goes for the rest of the buttons.

File: src/components/Facepile/Facepile.html [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+                  <div class="ms-Persona-secondaryText">Public Relations</div>
+                </div>
+              </div>
+              <button class="ms-PeoplePicker-resultAction js-selectedRemove"><i class="ms-Icon ms-Icon--x"></i></button>
+            </li>
+            <li class="ms-PeoplePicker-selectedPerson">
+              <div class="ms-Persona ms-Persona--sm">
+                <div class="ms-Persona-imageArea">
+                  <div class="ms-Persona-initials ms-Persona-initials--purple">ML</div>
+                  <img class="ms-Persona-image" src="">
```
Comment (battletoilet) [inline src/components/Facepile/Facepile.html]: It looks like all these image tags don't have an Alt attribute and also `src=""` cannot be empty, we should have a fallback image at least if we are waiting to load one in.

**Pull request merged**